core/DataAccess/Model.php

```diff
@@ -758,6 +764,13 @@ public function getNextInvalidatedArchive($idSite, $archivingStartTime, $idInval
             $archivingStartTime,
         ];
 
+        if ($onlySelectSegmentArchives) {
+            // we select only segment archiving by looking for archive done flags that have a length > 32. this works since the done flag
+            // for a segment archive will contain its hash
+            $sql .= " AND CHAR_LENGTH(name) > 32 AND date1 = ? AND date2 = ? AND period = ?";
```

Comment:
Regarding the `32` chars. Wouldn't that also select invalidations for plugins with long names. e.g. `SearchEngineKeywordsPerformance` actually already has 32 chars 🙈 
I'm not deep enough into the code here to be sure if that cause any issues or not.

Reply:
Yeah this could be an issue then. I guess we also have to do a split or something...